system/Database/BaseConnection.php

```diff
@@ -1269,8 +1269,12 @@ protected function _escapeString(string $str): string
      */
     public function callFunction(string $functionName, ...$params): bool
     {
-        $driver = strtolower($this->DBDriver);
-        $driver = ($driver === 'postgre' ? 'pg' : $driver) . '_';
+        $driver = strtolower($this->DBDriver) . '_';
+        if ($this->DBDriver === 'postgre') {
+            $driver = 'pg_';
+        } elseif ($this->DBDriver === 'oci8') {
+            $driver = 'oci_';
+        }
```

Comment:
This is the reason why the coverages are treated as being lowered.
What should I do?

It may be possible to fudge the coverage by doing the following, but it is a tricky code.

```php
$driver = strtolower($this->DBDriver);
$driver = (['postgre' => 'pg', 'oci8' => 'oci'][$driver] ?? $driver) . '_';
```

Reply:
Something like this?
```diff
@@ -1269,12 +1273,7 @@ abstract class BaseConnection implements ConnectionInterface
      */
     public function callFunction(string $functionName, ...$params): bool
     {
-        $driver = strtolower($this->DBDriver) . '_';
-        if ($this->DBDriver === 'postgre') {
-            $driver = 'pg_';
-        } elseif ($this->DBDriver === 'oci8') {
-            $driver = 'oci_';
-        }
+        $driver = $this->getDriverFunctionPrefix();
 
         if (strpos($driver, $functionName) === false) {
             $functionName = $driver . $functionName;
@@ -1291,6 +1290,11 @@ abstract class BaseConnection implements ConnectionInterface
         return $functionName(...$params);
     }
 
+    protected function getDriverFunctionPrefix()
+    {
+        return strtolower($this->DBDriver) . '_';
+    }
+
     //--------------------------------------------------------------------
     // META Methods
     //--------------------------------------------------------------------
```

Reply:
@kenjis 

Thanks, response.
I think it's really great.
You're a god.

Reply:
splited #5178